### PR TITLE
Did some improvements for leaf-test

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes # .

Changes proposed in this pull request:
- Added arr.ai test case sample code, like multiple test cases in one arr.ai test file,
- Make sure all tests can run in one go, even some tests are failed,
- Removed the info log in isRecursivelyTrue as it is tedious,
- Changed some format

Current test logs are:
```
~/anzx/arrai$ arrai test
Tests:
test/test1/io_test.arrai
test/test1_test.arrai
test/test2_test.arrai
test/test3_test.arrai

failed test: not true
value: {}

test/test1_test.arrai:1:3:
//test.assert.true(false)

failed test: not equal
expected: 42
actual:   112

test/test2_test.arrai:1:3:
//test.assert.equal(42)(16 * 7)
2/4 tests passed
2/4 tests passed
```
Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: internal/test/report.go [hunk 1]
```diff
@@ -7,17 +7,17 @@ import (
 
 // Report prints the results of the testing to the IO writer.
 func Report(w io.Writer, rs Results) error {
-	_, err := fmt.Fprintf(w, "%d/%d tests passed", rs.CountPassed(), rs.Count())
+	_, err := fmt.Fprintf(w, "%d/%d tests passed\n", rs.CountPassed(), rs.Count())
```
Comment (orlade-anz) [inline internal/test/report.go:10]: We don't need both this log and both the logs below. I think just remove this one.

**Review by orlade-anz — APPROVED** [review]

**Comment by orlade-anz** [comment]
Certainly an improvement, thanks for that :)

File: examples/test/multiple_cases_test.arrai [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+let testCase4 = (
+    //test.assert.true({"Hello", "arrai!"})
+);
+
+let testCase5 = (
+    //test.assert.false({})
+);
+
+let testCases = (:testCase1, :testCase2, :testCase3, :testCase4, :testCase5);
+testCases
```
Comment (marcelocantos) [inline examples/test/multiple_cases_test.arrai:27]: This is cleaner:

```arrai
(
    testCase1: //test.assert.equal("Hello, arrai!", //seq.join(", ", ["Hello", "arrai!"])),
    testCase2: //test.assert.unequal("Hello, arrai!", //seq.join(", ", ["Nihao", "arrai!"])),
    testCase3: //test.assert.size(2, {"Hello", "arrai!"}),
    testCase4: //test.assert.true({"Hello", "arrai!"}),
    testCase5: //test.assert.false({}),
)
```

Comment (ericzhang6222) [reply examples/test/multiple_cases_test.arrai:27]: Fixed in branch leaf-test.

**Pull request merged**